src/FieldWarning/Assets/Model/Game/Loading.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+
+public abstract class Loading
+{
+    private static Mutex mut = new Mutex();
+    public double percent_done = 0;
+    public bool finished = false;
+    public bool started = false;
+    public string name;
+
+    private List<Tuple<LoadingThreadDelegate,string>> funcs = 
+        new List<Tuple<LoadingThreadDelegate, string>>();
+    public Tuple<LoadingThreadDelegate, string> currentWorker = null;
+
+
+    public delegate void LoadingThreadDelegate();
+
+    public Loading(string name)
+    {
+        this.name = name;
+        LoadingScreen.workers.Enqueue(this);
+    }
+
+    public void AddWorker(LoadingThreadDelegate func, string text = "")
+    {
+        funcs.Add(new Tuple<LoadingThreadDelegate,string>(func,text));
+    }
+
+
+    public void Load()
```

Comment:
As far as I can tell these methods are only used in TerrainMap, so they should be at least protected (if you go with inheritance).

Reply:
actually this method Load() is used by LoadingScreen. It needs to be public. I can probably redesign it later to have lower visibility